content/applications/finance/payment_acquirers/ogone.rst

```diff
@@ -5,84 +5,105 @@ Ogone
-~~~~~~~~~~
+.. tip::
+   If you need to use another algorithm, such as `sha-256` or `sha-512`, within Odoo, activate the
+   :ref:`developer mode <developer-mode>` and go to :menuselection:`General Settings -->
+   Technical --> System Parameters`. From here, search for `payment_ogone.hash.function` and change
+   the value line to the desired algorithm (`sha-256` or `sha-512`).
 
-In order to retrieve the SHA Key IN, log into your ogone account, go to
-:menuselection:`Configuration --> Technical Information --> Data and origin verification -->
-Checks for e-Commerce & Alias Gateway`, and retrieve **SHA Key IN**.
+You are now required to generate **SHA-IN** passphrases. **SHA-IN** and **SHA-OUT** passphrases are
+used to digitally sign the transaction requests and responses between Odoo and Ogone. By using these
+secret passphrases and the `sha-1` algorithm, both systems can ensure that the information they
+receive from the other was not altered or tampered with.
 
-.. _ogone/sha_key_out:
+Enter the same **SHA-IN** passphrase in both :guilabel:`Checks for e-Commerce & Alias Gateway` and
+:guilabel:`Checks for DirectLink and Batch (Automatic)`. You can leave the IP address field blank.
 
-SHA Key OUT
-~~~~~~~~~~~
+Your **SHA-IN** and **SHA-OUT** passphrases should be different, and between 16 and 32 characters
+long. Make sure to use the same **SHA-IN** and **SHA-OUT** passphrases throughout the entire Ogone
+configuration, as Odoo only allows a single **SHA-IN** and single **SHA-OUT** passphrase.
 
-In order to retrieve the SHA Key OUT, log into your ogone account, go to
-:menuselection:`Configuration --> Technical Information --> Transaction feedback --> All transaction
-submission modes`, and get or generate your **API Key** and **Client Key**. Be careful to copy your
-API key as you'll not be allowed to get it later without generating a new one.
+In order to retrieve the **SHA-OUT** key, log into your Ogone account, go to
+:menuselection:`Configuration --> Technical --> Information --> Transaction feedback --> All
+transaction submission modes`, and get or generate your API Key and Client Key. Be careful to copy
+your API key as you’ll not be
 
-.. important::
-   If you are trying Ogone as a test, with the Test Account, change the **State** to *Test Mode*. We
-   recommend doing this on a test Odoo database, rather than on your main database.
+When done, head to :menuselection:`Configuration --> Technical Information --> Transaction Feedback`
+and check the following options:
 
-Configuration on Ogone
-======================
+- :guilabel:`I would like to receive transaction feedback parameters on the redirection URLs`:
+  should be checked.
+- :guilabel:`Direct HTTP server-to-server request`: should to be set to `Online but switch to a
+  deferred request when the online request fails`.
+- Both **URL** fields should contain the same following URL, with `example` replaced by your
+  database: `https://<example>/payment/ogone/return`.
```

Comment:
Either
```suggestion
- Both **URL** fields should contain the same following URL, with `example` replaced by your
  database: `https://example/payment/ogone/return`.
```
Or
```suggestion
- Both **URL** fields should contain the same following URL, with `<example>` replaced by your
  database: `https://<example>/payment/ogone/return`.
```
Otherwise people *will* put `https://<mydatabase>/...`